server/src/main/java/org/opensearch/index/mapper/StringFieldType.java

```diff
@@ -162,7 +186,7 @@ public Query wildcardQuery(String value, MultiTermQuery.RewriteMethod method, bo
         }
 
         Term term;
-        if (getTextSearchInfo().getSearchAnalyzer() != null) {
+        if (getTextSearchInfo().getSearchAnalyzer() != null && normalizeIfAnalyzed) {
```

Comment:
@nknize I think we still do have a conflict here: it makes sense to apply the `caseInsensitive` hint when user does not specify the analyzer to be used (in this case `getTextSearchInfo().getSearchAnalyzer() ` is the default one), but I think we should not do that if the `analyzer` or/and ` analyze_wildcard ` are provided 

[1] https://www.elastic.co/guide/en/elasticsearch/reference/7.10/query-dsl-query-string-query.html

Reply:
that's why QueryStringQueryParser [calls `normalizedWildcardQuery` method](https://github.com/opensearch-project/OpenSearch/pull/5462/files#diff-e6c8b3e3114a5b77343450a8810ecd8d1acdc772d92df51f71a2f89b4dca35d7R733). It ignores the `caseInsensitive` parameter.

Reply:
I think this is non issue: since you have this new method

```
public Query wildcardQuery(
        String value,
        MultiTermQuery.RewriteMethod method,
        boolean caseInsensitive,
        boolean normalizeIfAnalyzed,
        QueryShardContext context
    )
```

only `StringFieldType` could implement it, others won't. Another option I was thinking, may be we could embed this into `TextSearchInfo` / `MappedFieldType` as a property, since as you mentioned it is specific to `StringFieldType` only?

Reply:
What I mean is QueryStringQueryParser [calls the generic MappedFieldType.wildcardQuery](https://github.com/opensearch-project/OpenSearch/blob/2416d37a6570d6cfe6e7337a2afbba519d516a1e/server/src/main/java/org/opensearch/index/search/QueryStringQueryParser.java#L732) method. I can eliminate the normalized version of this method call by bumping the new method up the class hierarchy but I'd prefer not do that since `normalizeIfAnalyzed` is only relevant to the StringFIeldType. (and I don't like the idea of having to keep track of what the `caseInsensitive` and `normalizeIfAnalyzed` boolean logic means if we add another field type that doesn't care).

Reply:
Javadocs would help for sure, I don't think my suggestion (to have overloaded `wildcardQuery`)  is any better than new `normalizedWildcardQuery` - removes confusion in one place but hurts another one. What do you think about enriching `TextSearchInfo` for `StringFieldType`? (than we don't need this `normalizeIfAnalyzed` argument at all) I believe this could be exactly the right place to do so:

```
/**
 * Encapsulates information about how to perform text searches over a field
 *
 * @opensearch.internal
 */
public class TextSearchInfo {
  ...
}
```

Reply:
> What do you think about enriching TextSearchInfo for StringFieldType?

The problem is `QueryStringQueryParser` needs to change the normalization behavior at search runtime and TextSearchInfo is defined at index creation time for encapsulating Lucene fieldType parameters (e.g., norms, offsets, positions). This could be hacked around by adding an accessor to dynamically change based on the query but that changes the purpose of `TextSearchInfo` and I think adds some confusion on when to do this for what type of queries. It might be worth exploring but I think in a follow up enhancement issue since the surface area impacted is larger than the scope of this bug fix?